lexibank_chaconbaniwa.py

```diff
@@ -2,55 +2,76 @@
 import lingpy
 from clldutils.misc import slug
 from clldutils.path import Path
-from pylexibank.dataset import Concept
+from pylexibank import Concept
 from pylexibank.dataset import Dataset as BaseDataset
-from pylexibank.util import pb
+from pylexibank.util import progressbar
 
 
 @attr.s
-class BDConcept(Concept):
+class CustomConcept(Concept):
     Portuguese_Gloss = attr.ib(default=None)
 
 
 class Dataset(BaseDataset):
     dir = Path(__file__).parent
     id = "chaconbaniwa"
-    concept_class = BDConcept
-
-    def cmd_download(self, **kw):
-        pass
-
-    def cmd_install(self, **kw):
-
-        wl = lingpy.Wordlist(self.raw.posix("Bruzzi_Granadillo.txt"))
-
-        with self.cldf as ds:
-            ds.add_sources(*self.raw.read_bib())
-            for k in pb(wl, desc="wl-to-cldf"):
-                ds.add_language(
-                    ID=slug(wl[k, "doculect"]), Name=wl[k, "doculect"], Glottocode="bani1255"
-                )
-
-                ds.add_concept(
-                    ID=slug(wl[k, "concept"]),
-                    Name=wl[k, "concept"],
-                    Concepticon_ID=wl[k, "concepticon_id"] or "",
-                    Portuguese_Gloss=wl[k, "concept_portuguese"],
-                )
-
-                for row in ds.add_lexemes(
-                    Language_ID=slug(wl[k, "doculect"]),
-                    Parameter_ID=slug(wl[k, "concept"]),
-                    Value=wl[k, "entrj_in_source"],
-                    Form=wl[k, "ipa"],
-                    Segments=wl[k, "tokens"],
-                    Source=["granadillo_ethnographic_2006", "silva_discoteca_1961"],
-                ):
-                    cid = slug(wl[k, "concept"] + "-" + "{0}".format(wl[k, "cogid"]))
-                    ds.add_cognate(
-                        lexeme=row,
-                        Cognateset_ID=cid,
-                        Source=["Chacon2018"],
-                        Alignment=wl[k, "alignment"],
-                        Alignment_Source="Chacon2018",
-                    )
+    concept_class = CustomConcept
+
+    def cmd_makecldf(self, args):
+        # add sources
+        args.writer.add_sources()
+
+        # add languages
+        languages = args.writer.add_languages(lookup_factory="Name")
+
+        # add concepts
+        concepts = {}
```

Comment:
Okay, once you have the concepts now, one more thing (sorry, but this is to reduce code): First, name the column `Gloss` to `Name`, replace `_` by `-` in the `ID`, and then do:


```python
concepts = args.writer_add_concepts(id_factory=lambda x: x.split('-')[-1]+'_'+slug(x.Name), lookup_factory='Name')
```

I think this should work, right? I mean, if the concept list is then added to concepticon, one only needs to change the x.Name to x.english, and it's done...

Reply:
I had to change a bit to make it work in 94ad815. Would this be ok?

```python
        concepts = args.writer.add_concepts(
            id_factory=lambda cpt: "%s_%s" % (cpt.id.split("_")[0], slug(cpt.english)),
            lookup_factory="Name",
        )
```

There are no changes to the `cldf/` files.

Reply:
Yes. In any case, we'll profit from having less code there, right?